app/models/protocol_subscription.rb

```diff
@@ -142,15 +139,16 @@ def determine_streak(streak, current_response_completed, current_response_in_fut
     0
   end
 
-  def create_protocol_completion_entry(is_completed, is_periodical,
-                                       reward_points, is_in_future, streak, future_or_current)
+  def create_protocol_completion_entry(response, streak)
     result = {}
-    result[:completed] = is_completed
-    result[:periodical] = is_periodical
-    result[:reward_points] = reward_points
-    result[:future] = is_in_future
+    result[:completed] = response.is_completed
+    result[:periodical] = response.measurement.periodical?
+    result[:reward_points] = response.measurement.reward_points,
+    result[:future] = response.still_possible?
     result[:streak] = streak
-    result[:future_or_current] = future_or_current
+    result[:future_or_current] = response.future_or_current?
+    result[:open_from] = response.open_from
+    result[:protocol_name] = response.protocol.name
```

Comment:
```suggestion
    result[:questionnaire_key] = response.measurement.questionnaire.key
```
I don't think we need the protocol name because it will be the same for all entries in the protocol_completion. The questionnaire might differ, that's why we include it (to filter).

Reply:
oh, okay, yeah i'm not too familiar with the model and mixed them up, thanks!